### PR TITLE
Allowed functions in jest's describe

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
As of 22.0.5, jest allows classes for describe blocks.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/facebook/jest/blob/master/CHANGELOG.md#features-2
- [x] Increase the version number in the header if appropriate.

### CODE REVIEW

File: types/jest/index.d.ts [hunk 1]
```diff
@@ -199,6 +199,10 @@ declare namespace jest {
 
     type Lifecycle = (fn: ProvidesCallback, timeout?: number) => any;
 
+    interface IClassLike {
```
Comment (wsmd) [inline types/jest/index.d.ts]: What do you think about calling this interface `FunctionLike`?

Since classes are technically "special functions" and the `describe` implementation obviously works functions as well, I think `FunctionLike` would be better here for the sake of clarity.

I'm thinking about the following scenarios:

```js
const myInstance = new SomeClass();

describe(myInstance.constructor /* I don't know what you would do that but whatever 😅 */, () => {
  // ...
});
```

```js
function doSomething() {
  // ...
}

describe(doSomething, () => {
  // ...
});
```

Comment (JoshuaKGoldberg) [reply types/jest/index.d.ts]: SGTM! Made the change.

I also added `Function` to the allowed type union because some projects don't provide a `name` string in their `Function` typings (since it doesn't exist in IE).

**Comment by JoshuaKGoldberg** [comment]
Ehh, I don't think my changes are causing the failures... @RyanCavanaugh

File: types/jest/index.d.ts [hunk 2]
```diff
@@ -11,10 +11,9 @@
 //                 Jamie Mason <https://github.com/JamieMason>
 //                 Douglas Duteil <https://github.com/douglasduteil>
 //                 Ahn <https://github.com/AhnpGit>
+//                 Josh Goldberg <https://github.com/joshuakgoldberg>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.3
 
-declare var beforeAll: jest.Lifecycle;
```
Comment (AhnpGit) [inline types/jest/index.d.ts]: why did you remove this ?

File: types/jest/index.d.ts [hunk 3]
```diff
@@ -11,10 +11,9 @@
 //                 Jamie Mason <https://github.com/JamieMason>
 //                 Douglas Duteil <https://github.com/douglasduteil>
 //                 Ahn <https://github.com/AhnpGit>
+//                 Josh Goldberg <https://github.com/joshuakgoldberg>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.3
```
Comment (AhnpGit) [inline types/jest/index.d.ts]: Please put this back, Travis failed because it couldn't find this line.

**Review by ahnpnl — CHANGES_REQUESTED** [review]

File: types/jest/index.d.ts (hunk 3 above)
Comment (JoshuaKGoldberg) [reply types/jest/index.d.ts]: Spooky - must have been a bad merge...

**Review by wsmd — APPROVED** [review]

**Review by ahnpnl — APPROVED** [review]
